kernel/kernel-api/src/main/java/io/delta/kernel/statistics/DataFileStatistics.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.statistics;
+
+import static io.delta.kernel.internal.DeltaErrors.unsupportedStatsDataType;
+import static java.time.ZoneOffset.UTC;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoUnit.MILLIS;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.DeltaErrors;
+import io.delta.kernel.internal.util.JsonUtils;
+import io.delta.kernel.types.*;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/** Statistics about data file in a Delta Lake table. */
```

Comment:
I still think it is worth calling out a bit more how this class is used?

`Notably, this class it NOT used during data skipping -- that operations on columnar batches (or something, etc etc etc). Instead, this class is used for XXX YYY`